docs/recipes/upgrading/v4-to-v5.md

```diff
@@ -0,0 +1,257 @@
+	            // redirect requests that start with /api/ to the server on port 5000
+	            "/api/": {
+	                target: proxyTarget,
+	                changeOrigin: true,
+	            }
+	        }
+	    }
+	});
+	```
+
+1. Overwrite `src/Client/index.html` with the following content
+	
+	```html
+	<!doctype html>
+	<html>
+	<head>
+	    <title>SAFE Template</title>
+	    <meta charset="utf-8">
+	    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+	    <link rel="icon" type="image/png" href="/favicon.png"/>
+	    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.14.0/css/all.min.css">
+	</head>
+	<body>
+	    <div id="elmish-app"></div>
+	    <!-- Polyfill for remotedev -->
+	    <script type="text/javascript">
+	        var global = global || window;
+	    </script>
+	    <script type="module" src="/output/App.js"></script>
+	</body>
+	</html>
+	```
+	
+1. Run `npx tailwindcss init -p` in `src/Client`
+1. Change `tailwind.config.js` content with
+
+	```js
+	/** @type {import('tailwindcss').Config} */
+	module.exports = {
+	    mode: "jit",
+	    content: [
+	        "./index.html",
+	        "./**/*.{fs,js,ts,jsx,tsx}",
+	    ],
+	    theme: {
+	        extend: {},
+	    },
+	    plugins: []
+	}
+	```
+
+1. In order to get VSCode Tailwind intellisense, user needs to install `Tailwind CSS Intellisense`. We also create the file `.vscode/settings.json` with the following content.
+
+	*The regex here are for Feliz style DSL, if you want to support Fable.React DSL you need to adapt the regexes*
+	
+	```json
+	{
+	    "tailwindCSS.includeLanguages": {
+	        "fsharp": "html"
+	    },
+	    "tailwindCSS.experimental.classRegex": [
+	        "\\.className\\s+\"([^\"]*)\"",
+	        ["\\.className\\s+\\[([^\\]]*)\\]", "\"([^\"]*)\""]
+	    ]
+	}
+	```
+
+1. Create a file `src/Client/index.css` with the following content
+
+	```css
+	@tailwind base;
+	@tailwind components;
+	@tailwind utilities;
+	```
+	
+1. Add the following lines at the top of `src/Client/App.fs`
+	
+	```fs
+	open Fable.Core.JsInterop
+	
+	importSideEffects "./index.css"
+	```
+
+1. Overwrite `src/Client/Index.fs` with the new version of the file (see the diff)
+
+1. Create a file `tests/Client/vite.config.mts` with the following content:
+
+	```ts
+	import { defineConfig } from "vite";
+	
+	// https://vitejs.dev/config/
+	export default defineConfig({
+	    server: {
+	        port: 8081
+	    }
+	});
+	```
+	
+1. Overwrite `tests/Client/index.html` with 
+
+	```html
+	<!DOCTYPE html>
+	<html>
+	    <head>
+	        <title>SAFE.App Client Tests</title>
+	    </head>
+	    <body>
+	        <script type="module" src="/output/Client.Tests.js"></script>
+	    </body>
+	</html>
+	```
+	
+1. Create `.fantomasignore` with the following content:
+	
+	```
+	# Ignore Fable files
+	src/Client/output/fable_modules/ 
+	```
+
+1. Remove the existing `scripts` from the `package.json`, should be `"scripts": {},` now.
```

Comment:
Since we are editing the package.json here already, it could be good to bump the node version requirements as part of this step or as an adjacent step. i.e. set `engines.node = "~18 || ~20"`